src/Compilers/Test/Core/Assert/AssertEx.cs

```diff
@@ -160,6 +161,9 @@ public static void AreEqual<T>(T expected, T actual, string message = null, IEqu
             }
         }
 
+        public static void Equal<T>(ImmutableArray<T> expected, IEnumerable<T> actual)
```

Comment:
This change added `Assert.Equals<T>(T expected, T actual)`. This created an ambiguity with the following pattern in our code base:

```csharp
AssertEx.Equals([1, 2], someImmutableArray);
```

That call matches both the new overload and the existing overloads that have `ImmutableArray<T>` as a parameter. Unfortunately, the number of optional parameters is a tie breaker in overload resolution. That meant that the new overload was winning which is undesirable. Added the overloads without optional parameters to get the desired binding behavior.

Reply:
If that's the pattern, should we perhaps add `AssertEx.Equals<T>(ReadOnlySpan<T>, ImmutableArray<T>)` instead?

Reply:
Can you help me understand how that will help?

Reply:
Just avoid creating the immutable array for the expected value.

Reply:
This feeds into the equality that takes `IEnumerable<T>` though. I could rewrite that as well. IS that what you're suggesting?

Reply:
I was specifically going off of the pattern you showed above, which is `collectionExpression, ImmutableArray`. Is that not the pattern?